src/main/scala/edu/ie3/mobsim/model/ChargingBehavior.scala

```diff
@@ -74,39 +75,49 @@ object ChargingBehavior extends LazyLogging {
         logger.debug(
           s"${ev.getId} arrives at charging hub and wants to start charging."
         )
-        ev.getDestinationPoi.nearestChargingStations.keys.headOption
-          .map(_.getUuid)
+        (
+          ev.destinationPoi.nearestChargingStations.keys.headOption
+            .map(_.uuid),
+          None
+        )
       } else {
         /* Update charging prices memory of EV to have a reference for the prices of specific charging stations */
-        ev.getDestinationPoi.nearestChargingStations.keys.foreach { cs =>
+        val priceQueue: mutable.Queue[Double] = mutable.Queue.empty
```

Comment:
You can easily solve this in a functional style without using mutable objects :)
Hint: for that you'd probably need to call `map` instead of `foreach` below.